### DISCUSSION TITLE
Minimum valve opening value

### DISCUSSION THREAD

**Discussion by gbasile** [jmcollin78/versatile_thermostat#667] [Q&A] [answered]
I guess it's a flow/installation issue of my heating system, but one of my radiator does not heat at all if the valve opening value is less than 35%. 

I would like to understand if there is a way to configure a device so that it never goes below of this value.

**Comment by flg** [comment]
Hi,

AFAICT there is no such feature in Vtherm so it depends on your device. What is it and how do you interface to it?

**Reply by gbasile** [reply]
I have SONOFF TRVZB. 

At the moment I'm using them as valves thermostats either the beta releases. 

Given the fact that I have a central boiler I'm also considering to setup them as switch thermostat in order to minimise the time the gas heather is on.

**Reply by flg** [reply]
As far as I understand, with the new mode being developed (available in the betas), the attributes "Valve opening/closing degree" are under the control of Vtherm so I don't see how you could limit either of these without a new feature in vtherm.

**Reply by gbasile** [reply]
What do you think about setting up them as switches? 

On = 100% open
Off = 0% open

I understand that I will minimise flow problem and the time the boiler is on. 

I guess I might not have the same comfort as a valve smoothly closing as it reaches the target temperature

**Reply by flg** [reply]
I can't really comment as I've never tested that. One potential risk is draining the batteries if it frequently switch on and off for temperature regulation; but I guess you can tune the algorithm.

**Reply by jtulak** [reply]
Well, if you want to use the on/off behavior and you have Sonoff TRVZB, then all you need to do is to not use the valve control. Sonoff behaves like that by default. Just set target temperature at them and let the valve itself to do its logic.

But if I can have a little digress on the central gas boiler and on/off control in general: even 20 years old gas boilers I have seen are smart enough to regulate the amount of gass to stabilize water temperature and there are periods where very, very tiny flame is kept going. See these two charts - the lower one is water temperature on the returning side, and you can see that once it reaches some value, it just oscilates around it in a very small steps. The outgoing temperature is stable around 55°C.
<img width="893" alt="image" src="https://github.com/user-attachments/assets/08ea99a5-dbc6-469a-ba2b-91e11ce26b12" />
This is just the regular HA thermostat in a house with no smart TRVZB installed, just manual valves and one central boiler.

The problem with going on a full blast every time you turn it on, then shutting it off, then turning it on... is that it is actually wasting more gas than if it keeps puffing along. Just like in a car, driving gas-brake-gas-brake consumes more fuel than cruising stable.

**Comment by jmcollin78** [comment]
One thing that should work: create your own Number entity with min=35 and max=100.
Give it to VTherm as opening_degree. VTherm should not goes under 35% (the value is clamped to min/max).
With automation, listen to the changes of your number and send its value to the real opening_degree.

What is important here is that the number entity which represent the opening_degree has 35% as min.

**Reply by Riot2891** [reply]
@jmcollin78 @gbasile 
I face the same issue in my living room / dining room. Like jmcollin78 suggested creating a numeric helper is the way to go. But i strongly adwise you not to use a min state of 35 because the valve position in which water flows through your radiator is not static, it is influenced by the state of all the other radiators in your home. Therefore you want your valve to open min 35 when heating but close completely when not heating. Otherwise ther might be situations in whicht your radiator heats even when switched off.

I suggest you to do as i did:
1. Create a numeric helper ranging from min=0 to max=100.
2. Download my Blueprint  from github (which is based on a blueprint from arnie580): 
https://github.com/Riot2891/Home-Assistant-Blueprints/blob/main/TRVZB%20Offset.yaml

3. Create an automation using the blueprint.
5. In VTherm use the numeric Helper instead of the valve opening position. It will take care of valve opening an closing at the same time. If you need to provide an entity for valve closing in VTHerm, just create another numeric helper for valve closing but without any automation.

The automation works like this:
For any valve opening >0% it will add whatever number you selected as "Thermostat Offset" as long as the result <=100. If the result of the equation is larger than 100 it will just use 100. 
For example Vtherm sets valve opening to 5% and Thermostat Offset = 35, the automation will calculate 5+35=40%, on the other hand if VTherm trys to setthe valve to 70% it will set the valve to 100% as the result of the equation (70+35=105) is larger than 100.
If the result of the equation equals your chosen offset, it will set the valve opening position to 0 to close it completely. This will always be the case when VTherm sets the valve opening to 0 as 0+"Thermostat Offset" = "Thermostat Offset".
It will do the same for valve closing.

Hope this helps you out, let me know if you need further hlep.

**Reply by Riot2891** [reply]
Dear all, i updated my previous post. The blueprint now offers an input selector for the valve offset and is availible on from my github repositorie.

**Reply by jmcollin78** [reply]
Hello @Riot2891 

I think I can implement this directly on VTherm. This seems to be helpful for many people.

**Reply by jmcollin78** [reply, marked as answer]
https://github.com/jmcollin78/versatile_thermostat/issues/722

**Reply by Riot2891** [reply]
Hello @jmcollin78 
thanks for your reply, that would be realy nice,

**Reply by jmcollin78** [reply]
I am currently on it.

**Reply by jtulak** [reply]
@jmcollin78 Thank you for adding the minimum value. I'm just implementing a helper with the opposite max valve logic, to limit my too powerful heating panels from fully opening a valve even if there is a large temperature gradient. Would you kindly consider this feature too? 

I mean, Sonoff TRVZB has this feature built-in, but VTherm is using it to force specific valve position, so it makes sense to implement both sides.

**Reply by TallDaikon** [reply]
@jmcollin78 
Thanks. The feature is very helpful, but can still be optimized.
You always set the “Valve open degree” value to at least the minimum. As a result, the lower working range up to “minimum opening degree” is simply missing.
@Riot2891  adds “Valve open percentage” + “minimal opening degree” instead -> the upper working range is not controllable here.
Nevertheless, I think this approach is better.

Another option would be a conversion:

'Valve open degree' = ‘minimal opening degree’ + ‘Valve open degree ’* (100-”minimal opening degree” )/100

**Reply by jmcollin78** [reply]
Hello, understood and I think you are right. I wonder if it is a special use case or something to be generalized.

**Reply by Riot2891** [reply]
While in real life there should not be a difference as the valves we control are not designed to limit flow and therefore there is no linear correlation between valve open percentage and flow this is indeed a better way for the calculation.

**Reply by jmcollin78** [reply]
It has been implemented in release 7.1.0 release yesterday.
